com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs

```diff
@@ -245,20 +243,61 @@ internal set
 
         private ulong m_LocalClientId;
 
+        internal Dictionary<ulong, NetworkClient> connectedClients = new Dictionary<ulong, NetworkClient>();
+
+        internal List<NetworkClient> connectedClientsList = new List<NetworkClient>();
+
+        internal List<ulong> connectedClientIds = new List<ulong>();
+
         /// <summary>
-        /// Gets a dictionary of connected clients and their clientId keys. This is only populated on the server.
+        /// Gets a dictionary of connected clients and their clientId keys. This is only accessible on the server.
         /// </summary>
-        public readonly Dictionary<ulong, NetworkClient> ConnectedClients = new Dictionary<ulong, NetworkClient>();
+        public IReadOnlyDictionary<ulong, NetworkClient> ConnectedClients
+        {
+            get
+            {
+                if (IsServer == false)
+                {
+                    throw new NotServerException($"{nameof(ConnectedClients)} should only be accessed on server.");
+                }
+                return connectedClients;
+            }
+        }
 
         /// <summary>
-        /// Gets a list of connected clients. This is only populated on the server.
+        /// Gets a list of connected clients. This is only accessible on the server.
         /// </summary>
-        public readonly List<NetworkClient> ConnectedClientsList = new List<NetworkClient>();
+        public IReadOnlyList<NetworkClient> ConnectedClientsList
+        {
+            get
+            {
+                if (IsServer == false)
+                {
+                    throw new NotServerException($"{nameof(ConnectedClientsList)} should only be accessed on server.");
+                }
+                return connectedClientsList;
+            }
+        }
 
         /// <summary>
-        /// Gets a list of just the IDs of all connected clients.
+        /// Gets a list of just the IDs of all connected clients. This is only accessible on the server.
         /// </summary>
-        public ulong[] ConnectedClientsIds => ConnectedClientsList.Select(c => c.ClientId).ToArray();
+        public IReadOnlyList<ulong> ConnectedClientsIds
+        {
+            get
+            {
+                if (IsServer == false)
+                {
+                    throw new NotServerException($"{nameof(connectedClientIds)} should only be accessed on server.");
+                }
+                return connectedClientIds;
+            }
+        }
+
+        /// <summary>
+        /// Gets the local <see cref="NetworkClient"/> for this client.
+        /// </summary>
+        public NetworkClient LocalClient { get; internal set; }
```

Comment:
I see this getting set in connection approval, but I don't ever see it unset?

Reply:
Yes this was already an issue before where ConnectedClients was not cleared until the next initialize. I added a line which clears LocalClient in initialize. So this will behave differently as before which is not ideal behavior but I think we decided to touch the initialization/shutdown flow it should be in a separate PR.